arches/app/models/models.py

```diff
@@ -1360,14 +1360,24 @@ def __init__(self, *args, **kwargs):
     def is_fully_provisional(self):
         return bool(self.provisionaledits and not any(self.data.values()))
 
+    def set_missing_keys_to_none(self):
+        any_key_set = False
+        if not self.nodegroup_id:
+            return any_key_set
+        for node in self.nodegroup.node_set.all():
+            if node.datatype == "semantic":
+                continue
+            node_id_str = str(node.pk)
+            if not node_id_str in self.data:
```

Comment:
I think this reads much more naturally
```suggestion
            if node_id_str not in self.data:
```